modules/schedule_module/jsx/scheduleIndex.js

```diff
@@ -0,0 +1,565 @@
+        console.error(error);
+      });
+  }
+  /**
+   * Updates filter state
+   *
+   * @param {object} filter passed from FilterForm
+   */
+  updateFilter(filter) {
+    this.setState({filter});
+  }
+
+  /**
+   * Sets Filter to empty object
+   */
+  clearFilter() {
+    this.updateFilter({});
+    history.replaceState({}, '', '?');
+  }
+  /**
+   * Sets table data of tab by id
+   *
+   */
+  tabByid() {
+    const today = this.state.data.fieldOptions.today;
+    const next = this.state.data.fieldOptions.next30days;
+    const list = this.state.data.Data;
+    this.setState({tabledatapast: list.filter((e)=>e[7]<today)});
+    this.setState({tabledatanext: list.filter((e)=>{
+    return e[7]>today && e[7]<=next;
+    })});
+    this.setState({tabledatatoday: list.filter((e)=>e[7]==today)});
+  }
+  /**
+   * Store the value of the element in this.state.formData
+   *
+   * @param {string} formElement - name of the form element
+   * @param {string} value - value of the form element
+   */
+  setFormData(formElement, value) {
+    let formData = this.state.formData;
+    formData[formElement] = value;
+    if (formElement === 'DCCID') {
+    formData['PSCID'] = null;
+    this.fetchDataForm('DCCID', formData['DCCID']);
+    }
+    if (formElement === 'PSCID') {
+    formData['DCCID'] = null;
+    this.fetchDataForm('PSCID', formData['PSCID']);
+    }
+    this.setState({
+      formData: formData,
+    });
+  }
+  /**
+   * Handles the submission of the Add Schedule form
+   *
+   * @param {event} e - event of the form
+   */
+  handleSubmit(e) {
+    let formData = this.state.formData;
+    let formObject = new FormData();
+    for (let key in formData) {
+      if (formData[key] !== '') {
+        formObject.append(key, formData[key]);
+      }
+    }
+    formObject.append('edit', this.state.editModal);
+
+    fetch(this.props.submitURL, {
+      method: 'POST',
+      cache: 'no-cache',
+      credentials: 'same-origin',
+      body: formObject,
+    })
+    .then((resp) => {
+      if (resp.ok && resp.status === 200) {
+        swal.fire('Success!', 'Schedule added.', 'success').then((result) => {
+          if (result.value) {
+            this.closeModal();
+            this.fetchData();
+          }
+        });
+      } else {
+        resp.text().then((message) => {
+          swal.fire('Error!', message, 'error');
+        });
+      }
+    })
+    .catch((error) => {
+      console.error(error);
+    });
+  }
+    mapColumn(column, cell) {
+          return cell;
+  }
+  /**
+   * Handles the delete a Schedule
+   *
+   * @param {string} id - appointment id
+   */
+ deleteid(id) {
+    let deleteurl = loris.BaseURL + '/schedule_module/appointment/' + id;
+    fetch(deleteurl, {
+      method: 'DELETE',
+      cache: 'no-cache',
+      credentials: 'same-origin',
+    })
+    .then((resp) => {
+      if (resp.ok && resp.status === 200) {
+        swal.fire('Success!', 'Schedule deleted.', 'success').then((result) => {
```

Comment:
Delete function doesn't have an 'Are you sure?' warning

<img width="979" alt="Screen Shot 2020-07-20 at 4 13 02 PM" src="https://user-images.githubusercontent.com/17415878/87981947-0aa11e80-caa4-11ea-84e0-ed126fa4f24b.png">

Reply:
done